torch/_subclasses/fake_tensor.py

```diff
@@ -115,6 +115,19 @@ def get_schema_info(func):
     return torch._C._SchemaInfo(func._schema)  # type: ignore[attr-defined]
 
 
+# many of the prims do not at the moment model aliasing or strides,
+# but as an incremenetal step, enable aten-aten decomp. these decomps
+# are used for aot autograd tracing so we would like to unify and add
+# additional testing to them
+@functools.lru_cache(None)
+def aten_to_aten_decomp(func):
+    from torch._decomp import decomposition_table
+
+    decompositions = torch._decomp.decompositions
+    decomp_attrs = [getattr(decompositions, attr) for attr in dir(decompositions)]
```

Comment:
dir? 🤔

Reply:
do we not like dir ? alternatively i could define a register_decomposition in `torch?/_decomp/decompositions` that adds these ops to a list

Reply:
It's just weird to do it this way because, like, we explicitly have 

```
# None of these functions are publicly accessible; get at them
# from torch._decomps
__all__: List[str] = []
```

so like why aren't you getting the decomps from the decompositions table lol

Reply:
The answer being, that, you ONLY want the decomps that were registered inside torch._decomp.decompositions, as those "tend" to be aten-to-aten decompositions. But it seems more semantically correct for this to be some sort of attribute associated with any given decomp (you could even compute it automatically, by checking if the free variables of the code object reference any prims); after all, you're going to miss aten-to-aten decomps that just happen to be in torch._refs.

I can still be convinced that this is a good unblock, but the commentary on the function isn't enough to figure out the nuance.

Reply:
```
# many of the prims do not at the moment model aliasing or strides,
# but as an incremental step, enable aten-aten decomp. these decomps
# are used for aot autograd tracing so we would like to unify and add
# additional testing to them
``` 

was the comment - I think the misleading thing here is just the `aten-aten` name

Reply:
> so like why aren't you getting the decomps from the decompositions table lol
This seemed like the smallest change set and least invasive for what is supposed to be a somewhat temporary state. All of the decomps will eventually need to work 100% accurately for dynamic shapes, although we're a bit away from that at the moment. I'm happy to take a different route to accomplish that if you have another suggestion.

As for why I am enabling just these decomps - these are the decomps that are being enabled for torchinductor, not all of the prims, so they are not on the critical path for correctness with static shapes. For the ones that are (the ones I am using here), I think it's important to consolidate on their implementation, and add additional testing. For instance, it would have caught https://github.com/pytorch/torchdynamo/issues/1356.

Reply:
I'm not sure what's a better name than `aten_to_aten`. `non_prim_decomp` ?

Reply:
I guess, this isn't really something you resolve with naming; it's something you resolve with comments, and then socializing the comments with people. For example, we still have @mruberry telling folks that "eventually, we should move all torch._decomp to torch._refs". But this change is antithetical to that idea, since moving a decomp from here to refs prevents it from being exercised here. I don't think the idea here is fundamentally incompatible, but there is some needle to thread here.

Reply:
You could also name this exactly what it does on the tin: `torch_decomp_decompositions`